backend/src/main/java/atlantbh/restaurants/services/BaseService.java

```diff
@@ -0,0 +1,44 @@
+package atlantbh.restaurants.services;
+
+import atlantbh.restaurants.models.BaseModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public class BaseService<M extends BaseModel, R extends JpaRepository<M,Long>> {
+
+    @Autowired
+    protected R repository;
+
+    public Optional<M> getById(Long id) {
+        return repository.findById(id);
+    }
+
+    public void save(M m) {
```

Comment:
```
public M create(M model) throws ServiceException {
		try {
			repository.save(model);
			return model;
		} catch (RepositoryException e) {
			throw new ServiceException(e);
		}
	}
```
Create RepositoryException class, which extends Exception, and that will be your exception handling on repository level..

Reply:
Create update/delete methods, based on create example..
update -> pass (id, data) -> repository.findById (if there is no model, throw service excepiton) ->
call model.update(data) //**this is purpose of update model in base model** -> call repository.update(model); (catch repository exception if update fails and throw service exception).

Reply:
Fixed in c2ed9b4060983e1312e68ee9cdd1de2be1986207